router/src/matching/horizontal/static_segment.rs

```diff
@@ -15,10 +15,20 @@ impl PossibleRouteMatch for () {
     fn generate_path(&self, _path: &mut Vec<PathSegment>) {}
 }
 
+pub trait Routable: Debug + Clone {
+    fn into_str(&self) -> &'static str;
```

Comment:
I'd recommend `as_path()` as a name for this method: `as_` in line with the [Rust API Guidelines](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv) for a lightweight `&`-to-`&` conversion, and `as_path()` to give a little more context than `as_str()`.